cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/LogHandler.java

```diff
@@ -124,17 +124,21 @@ private void doGetLogs(HttpResponder responder, LoggingContext loggingContext,
       ReadRange readRange = new ReadRange(timeRange.getFromMillis(), timeRange.getToMillis(),
                                           LogOffset.INVALID_KAFKA_OFFSET);
       readRange = adjustReadRange(readRange, runRecord, fromTimeSecsParam != -1);
-
+      AbstractChunkedLogProducer logsProducer = null;
       try {
         // the iterator is closed by the BodyProducer passed to the HttpResponder
         CloseableIterator<LogEvent> logIter = logReader.getLog(loggingContext, readRange.getFromMillis(),
                                                                readRange.getToMillis(), filter);
-
-        AbstractChunkedLogProducer logsProducer = getFullLogsProducer(format, logIter, fieldsToSuppress, escape);
-        responder.sendContent(HttpResponseStatus.OK, logsProducer, logsProducer.getResponseHeaders());
+        logsProducer = getFullLogsProducer(format, logIter, fieldsToSuppress, escape);
       } catch (Exception ex) {
         LOG.debug("Exception while reading logs for logging context {}", loggingContext, ex);
+        if (logsProducer != null) {
+          logsProducer.close();
+        }
+        responder.sendStatus(HttpResponseStatus.INTERNAL_SERVER_ERROR);
```

Comment:
logic is different there, there we have a callable and we call close on any exception, which closes the channel. since its init is called (chunkStart) before any operation there, it will always return 200.

Even in here, if there were any exception during the read, we would have returned 200, as its chunked send. 

Only difference i see is, say if there was an exception while reading meta data, before reading any files. "/logs" endpoint would return 500, while "logs/next" would return 200. 

but i think that's not a critical fix at the moment, we can handle consistency later. @poornachandra what do you think?